tools/gen-modules.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
+
+# Generate genhdr/modules.h for inclusion in py/objmodule.c.
+
+from __future__ import print_function
+
+import sys
+import os
+from glob import glob
+
+def update_modules(path):
+    modules = []
+    for module in sorted(os.listdir(path)):
+        if not os.path.isfile('%s/%s/module.mk' % (path, module)):
+            continue # not a module
+        modules.append(module)
+
+    # Print header file for all external modules.
+    print('// Automatically generated by genmodules.py.\n')
+    for module in modules:
+        print('extern const struct _mp_obj_module_t %s_module;' % module)
```

Comment:
This also puts a restriction: the module struct in your .c file must be `<name>_module`.  That's worth documenting.

Also, to avoid possible clash with other names, it's probably worth making this name more descriptive, like `<name>_user_cmodule`.

Reply:
Done in the new commit.